homeassistant/components/prometheus/__init__.py

```diff
@@ -185,6 +189,44 @@ def handle_event(self, event):
         )
         last_updated_time_seconds.labels(**labels).set(state.last_updated.timestamp())
 
+    def handle_entity_registry_updated(self, event):
+        """Listen for deleted, disabled or renamed entities and remove them from the Prometheus Registry."""
+        if (action := event.data.get("action")) in (None, "create"):
+            return
+
+        entity_id = event.data.get("entity_id")
+        _LOGGER.debug("Handling entity update for %s", entity_id)
+
+        metrics_entity_id = None
+
+        if action == "remove":
+            metrics_entity_id = entity_id
+        elif action == "update":
+            changes = event.data.get("changes")
+            if "entity_id" in changes:
+                metrics_entity_id = changes["entity_id"]
+            elif "disabled_by" in changes or "name" in changes:
```

Comment:
For completion, should we also react on updates to `entity_category`?

Reply:
I don't think so. Since `entity_category` is not used in the labelsets, it shouldn't affect the metrics?

Reply:
Oh sorry, I meant `domain` rather than `entity_category`. I suspect we don't need to cover it since entities cannot move from one domain to another, right?